src/App.tsx

```diff
@@ -1,54 +1,108 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+
+const debounce = (func:React.Dispatch<React.SetStateAction<string>>,
+  delay:number) => {
+  let timerId: number;
+
+  return (...args:string[]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(func, delay, ...args);
+  };
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selected, setSelected] = useState<Person>();
+  const [showPeople, setshowPeople] = useState(false);
```

Comment:
```suggestion
  const [isDropDownActive, setIsDropDownActive] = useState(false);
```
'showPeople' sounds like a function name. Let's name the boolean variable according to these rules, so it's clear that it's a boolean
![image](https://github.com/mate-academy/react_autocomplete/assets/96409563/ce12c9be-2f2c-4f96-b0ce-772abfe14770)

https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig